src/annotator/components/test/NotebookModal-test.js

```diff
@@ -114,23 +122,66 @@ describe('NotebookModal', () => {
     assert.equal(document.body.style.overflow, 'hidden');
   });
 
-  it('hides modal on closing', () => {
-    const wrapper = createComponent();
+  context('when native modal dialog is not supported', () => {
+    let fakeDocument;
 
-    emitter.publish('openNotebook', 'myGroup');
-    wrapper.update();
+    beforeEach(() => {
+      fakeDocument = {
+        createElement: sinon.stub().returns({}),
+      };
+    });
 
-    let outer = wrapper.find(outerSelector);
-    assert.isFalse(outer.hasClass('hidden'));
+    it('does not render a dialog element', () => {
+      const wrapper = createComponent({}, fakeDocument);
 
-    act(() => {
-      wrapper.find('IconButton').prop('onClick')();
+      emitter.publish('openNotebook', 'myGroup');
+      wrapper.update();
+
+      assert.isFalse(wrapper.exists('dialog'));
     });
-    wrapper.update();
 
-    outer = wrapper.find(outerSelector);
+    it('hides modal on closing', () => {
+      const wrapper = createComponent({}, fakeDocument);
+
+      emitter.publish('openNotebook', 'myGroup');
+      wrapper.update();
+
+      let outer = wrapper.find(outerSelector);
+      assert.isFalse(outer.hasClass('hidden'));
+
+      act(() => {
+        wrapper.find('IconButton').prop('onClick')();
+      });
+      wrapper.update();
+
+      outer = wrapper.find(outerSelector);
+
+      assert.isTrue(outer.hasClass('hidden'));
+    });
+  });
+
+  context('when native modal dialog is supported', () => {
+    it('renders a dialog element', () => {
+      const wrapper = createComponent({});
 
-    assert.isTrue(outer.hasClass('hidden'));
+      emitter.publish('openNotebook', 'myGroup');
+      wrapper.update();
+
+      assert.isTrue(wrapper.exists('dialog'));
+    });
+
+    it('opens and closes native dialog', () => {
+      const wrapper = createComponent({});
+      const isDialogOpen = () => wrapper.find('dialog').getDOMNode().open;
+
+      act(() => emitter.publish('openNotebook', 'myGroup'));
+      wrapper.update();
+      assert.isTrue(isDialogOpen());
+
+      act(() => wrapper.find('IconButton').prop('onClick')());
```

Comment:
I think it would be better to give the button a semantic `data-testid`, for readability of this test and also in case we ever add other buttons to the content for some reason.